src/software/ai/hl/stp/play/offense/offense_play.cpp

```diff
@@ -26,20 +26,26 @@ void OffensePlay::getNextTactics(TacticCoroutine::push_type &yield, const World
 
 void OffensePlay::updateTactics(const PlayUpdate &play_update)
 {
-    PriorityTacticVector tactics_to_return;
-    unsigned int num_shoot_or_pass = play_update.num_tactics - 2;
-    unsigned int num_defenders     = 2;
     if (play_update.num_tactics == 0)
     {
         return;
     }
-    if (play_update.num_tactics <= 3)
+
+    PriorityTacticVector tactics_to_return;
+    unsigned int num_defenders     = 2;
+    unsigned int num_enemy_robots  = static_cast<int>(play_update.world.enemyTeam().numRobots());
+
+    if (num_enemy_robots <= 3)
+    {
+        num_defenders = num_enemy_robots > 1 ? 1 : 0;
```

Comment:
if they only have 1 robot we assume it's the goalie and we thus don't need any defenders

Reply:
ternary statements are highly discouraged